### PR TITLE
Change to Kotlin DSL for gradle and add publishing

### PR DESCRIPTION
**Pull request by incaseoftrouble**
Gradle now is in Kotlin and distribution is ready. regenerated / pruned verification metadata

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+
+            signing {
+                useGpgCmd()
+                sign(publishing.publications)
+            }
+
+            pom {
+                name.set("owl")
+                description.set("A tool collection and library for Omega-words, -automata and Linear Temporal Logic (LTL)")
+                url.set("https://github.com/incaseoftrouble/naturals-util")
```
Comment (sickert) [inline build.gradle.kts:436]:
```suggestion
                url.set("https://github.com/owl-toolkit/owl")
```

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+        distributionBaseName.set("owl-jre")
+
+        contents {
+            from("AUTHORS")
+            from("LICENSE")
+            from(compileMarkdown)
+
+            into("bin") {
+                from(buildKissat) {
+                    include("kissat") // Need more / less?
```
Comment (sickert) [inline build.gradle.kts:380]: LGTM

**Comment by incaseoftrouble** [comment]
> I think we can disable (comment out) the PMD check at the moment and saves us the time to address every warning.

The task itself takes <5 sec and currently only reports (doesn't fail the build - "ignore errors") I can also completely remove it?

**Comment by sickert** [comment]
> > I think we can disable (comment out) the PMD check at the moment and saves us the time to address every warning.
> 
> The task itself takes <5 sec and currently only reports (doesn't fail the build - "ignore errors") I can also completely remove it?

This is my mistake. I attributed the failing CI check to PMD without looking at it.

I think, if we don't plan to fix these warnings, then we should not run the analysis. Agreed?

**Comment by incaseoftrouble** [comment]
> I think, if we don't plan to fix these warnings, then we should not run the analysis. Agreed?

I think it makes sense to fix these issues, but maybe in a separate MR? Its ~180 issues (maybe some of them are superfluous and should be disabled?)

**Comment by incaseoftrouble** [comment]
The shfmt linter fails with an empty message, I don't know how to fix it.

File: .github/workflows/linter.yml [hunk 3]
```diff
@@ -31,3 +31,5 @@ jobs:
           VALIDATE_GOOGLE_JAVA_FORMAT: false
           # Too sensitive.
           VALIDATE_JSCPD: false
+          # No android in here
+          VALIDATE_KOTLIN_ANDROID: false
```
Comment (sickert) [inline .github/workflows/linter.yml:35]: I think we can simply disable the check for the files where the linting fails.

```suggestion
          VALIDATE_KOTLIN_ANDROID: false
          # Ignore files that are failed to be linted without error message
          FILTER_REGEX_EXCLUDE: <all broken sh files>
```

**Comment by sickert** [comment]
> > I think, if we don't plan to fix these warnings, then we should not run the analysis. Agreed?
> 
> I think it makes sense to fix these issues, but maybe in a separate MR? Its ~180 issues (maybe some of them are superfluous and should be disabled?)

Sounds good and I agree that we should disable some of the rules. I reviewed the PMD rule set and I think we should restrict the checks to the [Multithreading](https://pmd.github.io/pmd-6.45.0/pmd_rules_java_multithreading.html) and [Performance](https://pmd.github.io/pmd-6.45.0/pmd_rules_java_performance.html) set. The checks from the other sets seem to me opinionated and there are several justified exceptions in the code base to these rules.

**Comment by incaseoftrouble** [comment]
PMD yields different results on github actions compared to local execution, not sure what is happening there. I disabled pmd to be failing.

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+                    }
+                }
+
+                developers {
+                    developer {
+                        id.set("sickert")
+                        name.set("Salomon Sickert-Zehnter")
+                        email.set("salomon.sickert@mail.huji.ac.il")
+                        url.set("https://github.com/sickert")
+                        timezone.set("Asia/Teheran")
```
Comment (sickert) [inline build.gradle.kts:453]:
```suggestion
                        timezone.set("Asia/Jerusalem")
```

**Comment by sickert** [comment]
Thanks for the new build script!

**Comment by incaseoftrouble** [comment]
No problem. I could try to back-port the changes to 21.0 and push that version to maven central? Or is a new release on the horizon?

**Comment by sickert** [comment]
Yes, there is a new release on the horizon: I'm planning to do release for the next iteration of SYNTCOMP. Thus I don't think there is a need to backport the changes. Thanks for the offer.

**Pull request merged**